### ISSUE TITLE
Fine-tune animation distance modifiers to avoid skidding and running-in-place infantry

### ISSUE THREAD

**Issue by commy2** [TheSuperHackers/GeneralsGamePatch#2204]
There is an optional modifier after the Animation token in W3DModelDraw/ConditionState sub-modules.

```ini
Draw = W3DModelDraw ModuleTag_01
      ConditionState = MOVING (MOVING DAMAGED)
        Animation = [model animation name] [integer]
        ; Animation = DUDEANIM_SKN.DUDEANIM_RUN 32 ;<- the distance modifier
        ; there is an OPTIONAL number following the animation
        ; it represents the distance that a single loop of the animation would cover. 
        ; (ie, maybe it has the solder taking three strides, for a distance of 30 or so)
        ;
        ; how to tweak the number:
        ;   -- "skidding"?          reduce the number
        ;   -- "running in place"?  increase the number
```

This value can be tweaked to avoid skidding and running in place infantry. Someone should take the time and go through all infantry (and preferably also civilian) units and determine the right distance modifiers for nice looking walk cycle animations.

- [ ] AirF_AmericaInfantryColonelBurton
- [ ] AirF_AmericaInfantryMissileDefender
- [ ] AirF_AmericaInfantryPathfinder
- [ ] AirF_AmericaInfantryPilot
- [ ] AirF_AmericaInfantryRanger
- [ ] AmericaInfantryBiohazardTech
- [ ] AmericaInfantryCIAOfficer
- [ ] AmericaInfantryColonelBurton
- [ ] AmericaInfantryMissileDefender
- [ ] AmericaInfantryOfficer
- [ ] AmericaInfantryPathfinder
- [ ] AmericaInfantryPilot
- [ ] AmericaInfantryRanger
- [ ] AmericaInfantrySecretService
- [ ] AmericanFarmer01
- [ ] AsianFarmer01
- [ ] AsianFarmer02
- [ ] AsianFarmer3
- [ ] Boss_InfantryAngryMobNexus
- [ ] Boss_InfantryBlackLotus
- [ ] Boss_InfantryColonelBurton
- [ ] Boss_InfantryHacker
- [ ] Boss_InfantryJarmenKell
- [ ] Boss_InfantryPathfinder
- [ ] Boss_InfantryRanger
- [ ] Boss_InfantryTankHunter
- [ ] Chem_GLAInfantryAngryMobMolotov02
- [ ] Chem_GLAInfantryAngryMobNexus
- [ ] Chem_GLAInfantryAngryMobPistol01
- [ ] Chem_GLAInfantryAngryMobRock02
- [ ] Chem_GLAInfantryJarmenKell
- [ ] Chem_GLAInfantryRebel
- [ ] Chem_GLAInfantryStingerSoldier
- [ ] Chem_GLAInfantryTerrorist
- [ ] Chem_GLAInfantryTunnelDefender
- [ ] Chem_GLAInfantryWorker
- [ ] Chem_ToxicInfantryGamma
- [ ] ChinaAmbassador
- [ ] ChinaInfantryAgent
- [ ] ChinaInfantryBlackLotus
- [ ] ChinaInfantryHacker
- [ ] ChinaInfantryOfficer
- [ ] ChinaInfantryParadeRedGuard
- [ ] ChinaInfantryRedguard
- [ ] ChinaInfantrySecretPolice
- [ ] ChinaInfantryTankHunter
- [ ] Demo_GLAInfantryAngryMobMolotov02
- [ ] Demo_GLAInfantryAngryMobNexus
- [ ] Demo_GLAInfantryAngryMobPistol01
- [ ] Demo_GLAInfantryAngryMobRock02
- [ ] Demo_GLAInfantryHijacker
- [ ] Demo_GLAInfantryJarmenKell
- [ ] Demo_GLAInfantryRebel
- [ ] Demo_GLAInfantrySaboteur
- [ ] Demo_GLAInfantryStingerSoldier
- [ ] Demo_GLAInfantryTerrorist
- [ ] Demo_GLAInfantryTunnelDefender
- [ ] Demo_GLAInfantryWorker
- [ ] GC_Chem_GLAInfantryHijacker
- [ ] GC_Chem_GLAInfantryJarmenKell
- [ ] GC_Chem_GLAInfantryRebel
- [ ] GC_Chem_GLAInfantryStingerSoldier
- [ ] GC_Chem_GLAInfantryTerrorist
- [ ] GC_Chem_GLAInfantryTunnelDefender
- [ ] GC_Chem_GLAInfantryWorker
- [ ] GC_Chem_ToxicInfantryGamma
- [ ] GC_Slth_GLAInfantryHijacker
- [ ] GC_Slth_GLAInfantryJarmenKell
- [ ] GC_Slth_GLAInfantryRebel
- [ ] GC_Slth_GLAInfantrySniper
- [ ] GC_Slth_GLAInfantryStingerSoldier
- [ ] GC_Slth_GLAInfantryTerrorist
- [ ] GC_Slth_GLAInfantryTunnelDefender
- [ ] GC_Slth_GLAInfantryWorker
- [ ] GLAInfantryAngryMobMolotov02
- [ ] GLAInfantryAngryMobNexus
- [ ] GLAInfantryAngryMobPistol01
- [ ] GLAInfantryAngryMobRock02
- [ ] GLAInfantryAngryMobRock04
- [ ] GLAInfantryHijacker
- [ ] GLAInfantryJarmenKell
- [ ] GLAInfantryRebel
- [ ] GLAInfantrySaboteur
- [ ] GLAInfantryStingerSoldier
- [ ] GLAInfantryTerrorist
- [ ] GLAInfantryTunnelDefender
- [ ] GLAInfantryWorker
- [ ] GenericFemale01
- [ ] GenericFemale02
- [ ] GenericMale01
- [ ] GenericMale02
- [ ] HomelessGuy
- [ ] Infa_ChinaInfantryBlackLotus
- [ ] Infa_ChinaInfantryHacker
- [ ] Infa_ChinaInfantryMiniGunner
- [ ] Infa_ChinaInfantryTankHunter
- [ ] Lazr_AmericaInfantryColonelBurton
- [ ] Lazr_AmericaInfantryMissileDefender
- [ ] Lazr_AmericaInfantryPathfinder
- [ ] Lazr_AmericaInfantryPilot
- [ ] Lazr_AmericaInfantryRanger
- [ ] MogadishuFemaleCivilian01
- [ ] MogadishuFemaleCivilian02
- [ ] MogadishuMaleCivilian01
- [ ] MogadishuMaleCivilian02
- [ ] MogadishuMaleCivilian03
- [ ] Nuke_ChinaInfantryBlackLotus
- [ ] Nuke_ChinaInfantryHacker
- [ ] Nuke_ChinaInfantryRedguard
- [ ] Nuke_ChinaInfantryTankHunter
- [ ] Partisan01
- [ ] Partisan02
- [ ] Partisan03
- [ ] Slth_GLAInfantryAngryMobMolotov02
- [ ] Slth_GLAInfantryAngryMobNexus
- [ ] Slth_GLAInfantryAngryMobPistol01
- [ ] Slth_GLAInfantryAngryMobRock02
- [ ] Slth_GLAInfantryHijacker
- [ ] Slth_GLAInfantryJarmenKell
- [ ] Slth_GLAInfantryRebel
- [ ] Slth_GLAInfantrySaboteur
- [ ] Slth_GLAInfantryStingerSoldier
- [ ] Slth_GLAInfantryTerrorist
- [ ] Slth_GLAInfantryTunnelDefender
- [ ] Slth_GLAInfantryWorker
- [ ] SupW_AmericaInfantryColonelBurton
- [ ] SupW_AmericaInfantryMissileDefender
- [ ] SupW_AmericaInfantryPathfinder
- [ ] SupW_AmericaInfantryPilot
- [ ] SupW_AmericaInfantryRanger
- [ ] Tank_ChinaInfantryBlackLotus
- [ ] Tank_ChinaInfantryHacker
- [ ] Tank_ChinaInfantryRedguard
- [ ] Tank_ChinaInfantryTankHunter
- [ ] UNSoldier

**Comment by xezon** [comment]
Hmm this could take a lot of trial and error right?

**Comment by commy2** [comment]
Yes. We need an intern. Alternatively, you borrow the values from a mod, like NPM or Shockwave.

**Comment by xezon** [comment]
* Related to #1039

commy2
> Fixing the skidding / sliding thing should best be done with the diagonal movement fix. I tweaked an animation to look properly when moving cardinally, now it looks odd when walking diagonally.

* TheAssemblyArmada/Thyme#752
